web/src/features/VersionDiff/ViewDiffButton.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { Version } from "@types";
+import { useSelectedApp } from "@features/App/hooks/useSelectedApp";
+import { useIsHelmManaged } from "@components/hooks";
+
+interface Props {
+  onWhyNoGeneratedDiffClicked: (version: Version) => void;
+  onWhyUnableToGeneratedDiffClicked: (version: Version) => void;
+  onViewDiffClicked: (firstSequence: number, secondSequence: number) => void;
+  version: Version;
+  versionHistory: Version[];
+}
+
+interface DiffSummary {
+  filesChanged?: number;
+}
+
+// TODO: unmarshal this in the fetch handler
+function unmarshallDiffSummary(diffSummary: string): DiffSummary {
+  try {
+    return JSON.parse(diffSummary) || {};
+  } catch (err) {
+    throw err;
+  }
+}
+
+function getPreviousSequence(versionHistory: Version[], version: Version) {
+  let previousSequence = 0;
+  for (const v of versionHistory) {
+    if (v.status === "pending_download") {
+      continue;
+    }
+    if (v.parentSequence < version.parentSequence) {
+      previousSequence = v.parentSequence;
+      break;
+    }
+  }
+  return previousSequence;
+}
+
+function ViewDiffButton(props: Props) {
+  const { selectedApp } = useSelectedApp();
+  const { data: isHelmManagedResponse } = useIsHelmManaged();
+  const isHelmManaged = isHelmManagedResponse?.isHelmManaged || false;
+
+  // TODO: flatten in selector
+  const showViewDiffButton =
+    !isHelmManaged && !selectedApp?.downstream.gitops?.isConnected;
+  const showDiffSummaryError =
+    props.version?.diffSummaryError?.length > 0 ? true : false;
```

Comment:
change the name and made a boolean https://github.com/replicatedhq/kots/blob/d8ab2b5cc500cd296ab19fb82dc5f7ef05404196/web/src/components/apps/AppVersionHistory.tsx#L510-L511